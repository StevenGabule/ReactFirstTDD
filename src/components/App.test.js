import React from 'react';
import {configure, shallow} from 'enzyme';
import App from './App'
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});
const app = shallow(<App/>);

it('should renders app', () => {
    expect(app).toMatchSnapshot();
});

it('should initializes the state with an empty list of gifts', () => {
    expect(app.state().gifts).toEqual([]);
});

it('should add a new gift to state when clicking the add first button', () => {
    app.find('.btn-add').simulate('click');
    expect(app.state().gifts).toEqual([{id: 1}])
});
