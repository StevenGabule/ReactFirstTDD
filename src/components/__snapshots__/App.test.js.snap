// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders app 1`] = `ShallowWrapper {}`;
